homeassistant/components/switch/zigate.py

```diff
@@ -0,0 +1,124 @@
+"""
+ZiGate platform.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/ZiGate/
+"""
+import logging
+from homeassistant.components.switch import SwitchDevice
+
+DOMAIN = 'zigate'
+DATA_ZIGATE_DEVICES = 'zigate_devices'
+DATA_ZIGATE_ATTRS = 'zigate_attributes'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZiGate sensors."""
+    if discovery_info is None:
+        return
+
+    z = hass.data[DOMAIN]
+    import zigate
+
+    def sync_attributes():
+        devs = []
+        for device in z.devices:
+            actions = device.available_actions()
+            if not actions:
+                continue
+            for endpoint, action_type in actions.items():
+                if [zigate.ACTIONS_ONOFF] == action_type:
+                    key = '{}-{}-{}'.format(device.addr,
+                                            'switch',
+                                            endpoint
+                                            )
+                    if key not in hass.data[DATA_ZIGATE_ATTRS]:
+                        _LOGGER.debug(('Creating switch '
+                                       'for device '
+                                       '{} {}').format(device,
+                                                       endpoint))
+                        entity = ZiGateSwitch(device, endpoint)
+                        devs.append(entity)
+                        hass.data[DATA_ZIGATE_ATTRS][key] = entity
+
+        add_devices(devs)
+    sync_attributes()
+    zigate.dispatcher.connect(sync_attributes,
+                              zigate.ZIGATE_ATTRIBUTE_ADDED, weak=False)
+
+
+class ZiGateSwitch(SwitchDevice):
+    """Representation of a ZiGate switch."""
+
+    def __init__(self, device, endpoint):
+        """Initialize the ZiGate switch."""
+        self._device = device
+        self._endpoint = endpoint
+        self._name = 'zigate_{}_{}_{}'.format(device.addr,
+                                              'switch',
+                                              endpoint)
+        self._unique_id = '{}-{}-{}'.format(device.addr,
+                                            'switch',
+                                            endpoint)
+        self.registry_name = '{} {}'.format(device, endpoint)
+
+    @property
+    def unique_id(self)->str:
```

Comment:
Please move common properties and methods to a common base zigate device class that we create in the component module.